Classes/Core/Testbase.php

```diff
@@ -164,29 +166,121 @@ public function removeOldInstanceIfExists($instancePath): void
 
     /**
      * Link TYPO3 CMS core from "parent" instance.
+     *
      * For functional and acceptance tests.
      *
      * @param string $instancePath Absolute path to test instance
-     * @throws Exception
-     * @return void
+     * @throws \InvalidArgumentException
+     * @throws \RuntimeException
      */
-    public function setUpInstanceCoreLinks($instancePath): void
+    public function setUpInstanceCoreLinks(string $instancePath): void
     {
+        $filesystem = new Filesystem();
+
+        if (!$filesystem->isAbsolutePath($instancePath)) {
+            throw new \InvalidArgumentException(
+                sprintf('$instancePath (%s) must be an absolute path.', $instancePath),
+                1636244396
+            );
+        }
+
+        $this->defineOriginalRootPath();
+
+        /** @var array<string, string> $linksToSet */
         $linksToSet = [
-            '../../../../' => $instancePath . '/typo3_src',
-            'typo3_src/typo3' => $instancePath . '/typo3',
-            'typo3_src/index.php' => $instancePath . '/index.php',
+            ORIGINAL_ROOT => $instancePath . '/typo3_src',
+            'typo3_src/typo3/sysext/' => $instancePath . '/typo3/sysext',
```

Comment:
Nothing we can tackle with this change, but this code will have a hard time when typo3 extensions are installed in vendor ...